### PR TITLE
Patch/syntax and acl updates

### PR DESCRIPTION
**Pull request by jclausen**
This pull requests adds a new function `setAccessControlPolicy` and handles some cleanup of parameters which were being passed as resource  strings ( which were then being encoded and blowing up ).

`getBucket` has been updated to use the ListObjectsv2 API - which is recommended by AWS for more detailed information.

This PR is required for an upcoming PR to `cbfs` which will introduce an S3 disk provider.

### CODE REVIEW

**Comment by lmajano** [comment]
@jclausen I am currently reviewing this. Can you do me a favor. Can you link up to the s3 docs on amazon on the new api docs in the readme? This will help for ;future implementations and a nice reference.  Thanks

File: models/AmazonS3.cfc [hunk 1]
```diff
@@ -351,35 +351,62 @@ component accessors="true" singleton {
...
-	 *
-	 * @bucketName The bucket to get the ACL.
-	 * @uri        An optional resource uri to get the ACL.
-	 *
-	 * @return     An array containing the ACL for the given resource.
-	 */
-	array function getAccessControlPolicy( required string bucketName = variables.defaultBucketName, string uri = "" ) {
+     * Sets a bucket's or object's ACL policy.
+     *
+     * @bucketName The bucket to get the ACL.
```
Comment (lmajano) [inline models/AmazonS3.cfc]: Change to `The bucket to set the ACL` thanks

File: models/AmazonS3.cfc [hunk 2]
```diff
@@ -351,35 +351,62 @@ component accessors="true" singleton {
...
+     * @uri        An optional resource uri to get the ACL.
```
Comment (lmajano) [inline models/AmazonS3.cfc]: Same as above, change `get` to `set`

File: models/AmazonS3.cfc [hunk 3]
```diff
@@ -351,35 +351,62 @@ component accessors="true" singleton {
...
+	 * @acl        A known ACL string
+     *
+     */
```
Comment (lmajano) [inline models/AmazonS3.cfc]: Would be great to add a `@see https://link.to.amzon/acl/policies` here. So we can have that on the generated docs.

File: models/AmazonS3.cfc [hunk 4]
```diff
@@ -400,12 +427,14 @@ component accessors="true" singleton {
 		string maxKeys             = "",
 		string delimiter           = variables.defaultDelimiter
 	) {
-		requireBucketName( arguments.bucketName );
-		var parameters = {};
+        requireBucketName( arguments.bucketName );
+        var parameters = {
+			"list-type" : 2
```
Comment (lmajano) [inline models/AmazonS3.cfc:426]: Can you put a link here in a comment to this API Doc. Would be great to have it referenced

File: models/AmazonS3.cfc [hunk 5]
```diff
@@ -415,6 +444,10 @@ component accessors="true" singleton {
 			parameters[ "max-keys" ] = arguments.maxKeys;
 		}
 
+        if ( isNumeric( arguments.maxKeys ) ) {
```
Comment (lmajano) [inline models/AmazonS3.cfc]: This is repeated above?

**Review by lmajano — CHANGES_REQUESTED** [review]
Few updates, and it looks ready to go

File: models/AmazonS3.cfc (hunk 1 above)
Comment (jclausen) [reply models/AmazonS3.cfc]: Doh! Oops.

File: models/AmazonS3.cfc (hunk 5 above)
Comment (jclausen) [reply models/AmazonS3.cfc]: It is.  Looks like a bad merge.

File: models/AmazonS3.cfc (hunk 4 above)
Comment (jclausen) [reply models/AmazonS3.cfc:426]: I've placed this in the `@see` in the doc blocks

**Comment by jclausen** [comment]
All PR feedback should be resolved and this is ready to merge

**Pull request merged**